test/e2e/cross_os/test_job_submissions.py

```diff
@@ -0,0 +1,281 @@
+                },
+                {
+                    "onEnter": {
+                        "command": (
+                            "/bin/sleep" if get_operating_system_name() == "linux" else "timeout"
+                        ),
+                        "args": ["30"],
+                        "cancelation": {
+                            "mode": "NOTIFY_THEN_TERMINATE",
+                            "notifyPeriodInSeconds": 1,
+                        },
+                    },
+                },
+                "envEnter",
+            ),
+            (
+                {
+                    "onRun": {
+                        "command": "whoami",
+                    },
+                },
+                {
+                    "onEnter": {
+                        "command": "whoami",
+                    },
+                    "onExit": {
+                        "command": (
+                            "/bin/sleep" if get_operating_system_name() == "linux" else "timeout"
+                        ),
+                        "args": ["30"],
+                        "cancelation": {
+                            "mode": "NOTIFY_THEN_TERMINATE",
+                            "notifyPeriodInSeconds": 1,
+                        },
+                    },
+                },
+                "envExit",
+            ),
+        ],
+    )
+    def test_job_reports_canceled_session_action(
+        self,
+        deadline_resources,
+        deadline_client: DeadlineClient,
+        run_actions: Dict[str, Any],
+        environment_actions: Dict[str, Any],
+        expected_canceled_action: str,
+    ) -> None:
+        job = Job.submit(
+            client=deadline_client,
+            farm=deadline_resources.farm,
+            queue=deadline_resources.queue_a,
+            priority=98,
+            template={
+                "specificationVersion": "jobtemplate-2023-09",
+                "name": f"jobactioncancel-{expected_canceled_action}",
+                "steps": [
+                    {
+                        "name": "Step0",
+                        "script": {
+                            "actions": run_actions,
+                        },
+                    },
+                ],
+                "jobEnvironments": [
+                    {
+                        "name": "environment",
+                        "script": {
+                            "actions": environment_actions,
+                        },
+                    }
+                ],
+            },
+        )
+
+        def is_job_started(current_job: Job) -> bool:
+            current_job.refresh_job_info(client=deadline_client)
+            LOG.info(f"Waiting for job {current_job.id} to be created")
+            return current_job.lifecycle_status != "CREATE_IN_PROGRESS"
+
+        retries_for_job_start: int = 10
+        for i in range(retries_for_job_start):
+            if is_job_started(job):
+                break
+            sleep(10)
```

Comment:
This could be simplified if we wrapped the logic to call `is_job_started` in an inner function and used the `backoff` libraries annotations to manage the retrying logic